docs/tutorials/broadcast-join.rst

```diff
@@ -0,0 +1,298 @@
+    data store is used. Other supported data stores may be used by simply adapting the above parameters appropriately.
+
+Then we make use of Spark's ``DataFrameReader`` and our ``SpatialRDDProvider`` to create a ``DataFrame`` with geospatial
+types.
+
+.. code-block:: scala
+
+    val fipsDF = spark.read.format("geomesa")
+      .options(fipsParams)
+      .option("geomesa.feature", "fips")
+      .load()
+
+    val gdeltDF = spark.read.format("geomesa")
+      .options(gdeltParams)
+      .option("geomesa.feature", "gdelt")
+      .load()
+
+
+Filtering DataFrames (Optional)
+-------------------------------
+
+Depending on the scale of the data in our data store, and how specific our questions are, we may want to narrow the result
+before joining. For example, if we only wanted GDELT events within a one-week span, we could filter the DataFrame as
+follows:
+
+.. code-block:: scala
+
+    import spark.implicits._
+    val filteredGdelt = gdeltDF.where("dtg between '2018-01-01 12:00:00' and '2018-01-08 12:00:00'")
+
+Broadcast Join
+--------------
+
+Now we're ready to join the two data sets. This is where we will make use of our geospatial UDFs. ``st_contains`` takes
+two geometries as input, and it outputs whether the second geometry lies within the first one. For more documentation
+and a full list of the UDFs provided by GeoMesa see :doc:`/user/spark/sparksql_functions`.
+
+Using these two UDFs, we can build the following join query.
+
+.. code-block:: scala
+
+    import org.apache.spark.sql.functions.broadcast
+    val joinedDF = gdeltDF.join(broadcast(fipsDF), st_contains($"the_geom", $"geom"))
+
+The above query executes the broadcast as described previously, sending the FIPS data to each of the executors, then
+joining the two data sets based on whether the GDELT event occurred in the county.
+
+Aggregating
+-----------
+
+Now we have a DataFrame where each GDELT event is paired with the US county where it occurred.
+To turn this into meaningful statistics about the distribution of GDELT events in the US, we
+can do a ``GROUP BY`` operation and use some of SparkSQL's aggregate functions.
+
+.. code-block:: scala
+
+    import org.apache.spark.sql.functions.concat
+    import org.apache.spark.sql.functions.count
+    import org.apache.spark.sql.functions.first
+    val aggregateDF = joinedDF.groupBy(concat($"STATEFP", $"COUNTYFP"))
+      .agg(first("NAME").as("name"),
+           count($"globalEventId").as("eventCount"),
+           first("the_geom").as("geom"))
+
+The above query groups the data based on FIPS code, (which is split into a state and county code),
+and counts the number of distinct GDELT events in each one. The result can be used to generate a visualization
+of the event density in each county, which we will see in the next section.
+
+Visualization
+-------------
+
+To visualize this result, we first need to map our data to a format that most geospatial visualization tools prefer,
```

Comment:
I'd avoid the claim that most visualization tools prefer GeoJSON.  We can just say that we're going to render the data as GeoJSON.